fg/apps/main/views/download.py

```diff
@@ -171,3 +172,166 @@ def download_distribution_csv(request, uuid):
+            part['tags'] = tags
+            part['author'] = author
+            part['collections'] = []
+            sample['part'] = part             
+            sample['derived_from'] = None # might not be included in export
+            well = WellSerializer(well).data
+            well['sample'] = sample
+            wells.append(well)
+
+        # Add the wells back to the plate, remove container and protocol
+        plate = PlateSerializer(plate).data
+        plate['wells'] = wells
+        plate['container'] = None
+        plate['protocol'] = None
+        plate['plateset'] = pset
+        plate['distribution'] = dist
+        data.append(plate)
+
+    return data
+
+def generate_plate_json_response(plates, filename, distribution=None, plateset=None, content_type='application/json'):
+    '''a wrapper to generate the response, in case we need to modify the data
+
+       Parameters
+       ==========
+       plates: a list of plates to include (can be from a plateset, a single
+               plate, or a distribution)
+       filename: the complete filename to download to
+       content_type: the content type (defaults to application/json)
+       distribution: if provided, add the distribution to each plate.
+       plateset: if provided, include with distribution (to add plate to)
+    '''
+    data = generate_plate_json(plates, distribution, plateset)
+    response = HttpResponse(json.dumps(data, indent=4), content_type=content_type)
+    response['Content-Disposition'] = 'attachment; filename="%s"' % filename
+    return response
+
+
+@ratelimit(key='ip', rate=rl_rate, block=rl_block)
+def export_plate_json(request, uuid):
+    '''export json for a single plate'''
+    try:
+        plate = Plate.objects.get(uuid=uuid)
+
+        # Add the date, number of wells
+        filename = "freegenes-plate-%s-wells-%s.json" %(datetime.now().strftime('%Y-%m-%d'),
+                                                       plate.wells.count())
+        return generate_plate_json_response([plate], filename)
+    except Plate.DoesNotExist:
+        pass
+
+
+@ratelimit(key='ip', rate=rl_rate, block=rl_block)
+def export_plateset_json(request, uuid):
+    '''export json for a plateset (only the first)'''
+    try:
+
+        plateset = PlateSet.objects.get(uuid=uuid)
+        filename = "freegenes-plateset-%s-plates-%s.json" %(datetime.now().strftime('%Y-%m-%d'),
+                                                           plateset.plates.count())
+
+        if plateset.plates.count() > 0:
+            return generate_plate_json_response(plateset.plates.all(), plateset=plateset, filename=filename)
+
+        # No plates returns an empty csv
+        return generate_plate_json([], filename)
+
+    except PlateSet.DoesNotExist:
+        pass
+
+
+@ratelimit(key='ip', rate=rl_rate, block=rl_block)
+def export_distribution_json(request, uuid):
+    '''generate a json export for an entire distribution (include all plates)
+    '''
```

Comment:
I don't understand what you are saying here, can you re-phrase? The idea is that if there are two instances of a BioNet node, and a distribution is desired to be export from one (which comes down to a plateset and plates) we would want to be able to import into the other, and with confidence say "These two are the same distributions, the same platesets." Otherwise, you in fact are importing the same plates, but then calling them something different. If a third user then says "I want to find Distribution X" they can't search for the same identifier across nodes. It makes the "future Bionet Network" more complicated to remove the distribution /plateset export, but if that's what you want, sure.

Reply:
> with confidence say "These two are the same distributions, the same platesets."

> Otherwise, you in fact are importing the same plates, but then calling them something different.

We definitely want people to be able to find Distribution X across the network with the same identifiers, but that doesn't mean the same plates get exported everywhere. Plates can only physically exist in one place at a time, and so even though we want to export / import distributions and platesets, we don't want to export ALL plates, because only a subset of them will be shipped. It doesn't make sense for, say, Biotech Without Borders, to have 48 pOpen_v3 plates stored in their system, because in reality, they only have 2.

Reply:
okay understood, so we need an entire new view then.

Reply:
New view is added, need to do other stuff.